src/components/Forum/ForumApprovedPosts.js

```diff
@@ -0,0 +1,81 @@
+import React, { useMemo } from 'react';
+import { Route, Switch } from 'react-router-dom';
+import { useTheme } from '@material-ui/core';
+import DeleteIcon from '@material-ui/icons/Delete';
+import EditIcon from '@material-ui/icons/Edit';
+import ReplyIcon from '@material-ui/icons/Reply';
+import { ForumPostsListByApproval, FormattedReferencedDataField, RepliesChip } from './ForumPostsList';
+
+const approvedPostsListColumns = [
+  {
+    id: 'userID',
+    label: 'User',
+    format: (value) => (
+      <FormattedReferencedDataField
+        collection="users"
+        id={value}
+        field="displayName"
+        notFoundMessage="User not found"
+        render={(item) => <>{item}</>}
+      />
+    ),
+  },
+  {
+    id: 'title',
+    label: 'Post Title',
+  },
+  {
+    id: 'categoryID',
+    label: 'Category',
+    format: (value) => (
+      <FormattedReferencedDataField
+        collection="forum_categories"
+        id={value}
+        field="title"
+        notFoundMessage="Category not found"
+        render={(item) => <>{item}</>}
+      />
+    ),
+  },
+  {
+    id: 'approvedAt',
+    label: 'Approved At',
+    format: (value) => (value ? value.toDate().toLocaleString() : 'Date not found'),
+  },
+  {
+    id: 'id',
+    label: 'Replies',
+    format: (value) => <RepliesChip postID={value} />,
+  },
+];
+
+export default function ForumApprovedPostsList() {
+  const theme = useTheme();
+
+  const approvedActionButtons = useMemo(() => ([
+    {
+      Icon: ReplyIcon,
+    },
+    {
+      Icon: EditIcon,
+      color: theme.palette.text.main,
+    },
+    {
+      Icon: DeleteIcon,
+      color: theme.palette.error.main,
+    },
+  ]), [theme.palette.error.main, theme.palette.text.main]);
+
+  return (
+    <Switch>
+      {/* Nested route to an individual post goes here */}
+      <Route exact>
+        <ForumPostsListByApproval
+          approved
+          columns={approvedPostsListColumns}
+          actionButtons={approvedActionButtons}
+        />
+      </Route>
+    </Switch>
```

Comment:
`ForumApprovedPosts.js` doesn't seem to add much logic over the `ForumPostsListByApproval` component, would it make more sense to have the columns & icons arrays in `ForumHomePage.js`?

Reply:
It's not doing much right now, but I wanted to separate out the components because the functionality for approved posts is gonna get complex when we add the pending and approved replies and other components. On the other hand, the functionality for pendings is gonna be relatively simple. Let me know what you think.

Reply:
Ok yeah that makes sense if the functionality for approved posts is going to get much more complex, and I think the pendings being in a separate component w/o much added functionality is fine if the approved are also in its own component.